agent/wptdriver/webpagetest.cc

```diff
@@ -161,6 +203,61 @@ bool WebPagetest::GetTest(WptTestDriver& test) {
 }
 
 /*-----------------------------------------------------------------------------
+Call just before rebooting. If return false, reboot is cancelled
+-----------------------------------------------------------------------------*/
+bool WebPagetest::RebootWatchDog() {
+  short reboots = LoadRebootWatchDog();
+  if (reboots >= MAX_REBOOT) {
+    return false;
+  }
+  SaveRebootWatchDog(reboots + 1);
+  return true;
+}
+
+/*-----------------------------------------------------------------------------
+Reset reboot count
+-----------------------------------------------------------------------------*/
+void WebPagetest::ResetRebootWatchDog() {
+  SaveRebootWatchDog(0);
+}
+
+void WebPagetest::SaveRebootWatchDog(short nb) {
+  CString val;
+  val.Format(_T("%d"), nb);
+  HANDLE file_handle = CreateFile(_T("reboot_count.txt"), GENERIC_WRITE, 0, 0,
+    CREATE_ALWAYS, 0, 0);
+  if (file_handle != INVALID_HANDLE_VALUE) {
+    DWORD bytes;
+    WriteFile(file_handle, val, val.GetLength(), &bytes, 0);
+    CloseHandle(file_handle);
+  }
+}
+
+short WebPagetest::LoadRebootWatchDog() {
+  byte ReadBuffer[1] = { 0 };
+  short nb;
+
+  HANDLE file_handle = CreateFile(_T("reboot_count.txt"), GENERIC_READ, FILE_SHARE_READ,
+    0, OPEN_EXISTING, 0, 0);
+
+  DWORD bytes;
+  if (file_handle != INVALID_HANDLE_VALUE) {
+    // something bad is happening. Make sure we don't restart
+    nb = SHRT_MAX;
+  } else if (ReadFile(file_handle, ReadBuffer, sizeof(ReadBuffer), &bytes, 0) && bytes) {
+    // convert from ASCII
+    short nb = ReadBuffer[0] - 48;
+    AtlTrace(_T("[wptdriver] - Reboot count %d"), nb);

```

Comment:
Why AtlTrace instead of WptTrace? The latter will go to the log file, but the former, I am not sure.